src/MassTransit/Initializers/PropertyConverters/ListPropertyConverter.cs

```diff
@@ -64,14 +65,28 @@ public Task<List<TElement>> Convert<TMessage>(InitializeContext<TMessage> contex
                     return Task.FromResult(input.ToList());
             }
         }
+
+        Task<ICollection<TElement>> IPropertyConverter<ICollection<TElement>, IEnumerable<TElement>>.Convert<T>(InitializeContext<T> context, IEnumerable<TElement> input)
+        {
+            switch (input)
+            {
+                case null:
+                    return TaskUtil.Default<ICollection<TElement>>();
```

Comment:
I wouldn't really cache this, caching small objects worse than creating multiple:
```csharp
Task.FromResult<ICollection<TElement>>(Array.Empty<TElement>())
```

Reply:
but I'm curious what @phatboyg thinks